window.lisp

```diff
@@ -683,10 +683,17 @@ and bottom_end_x."
              (let ((key (copy-structure key)))
                (setf (key-shift key) t)
                key))
+           (key-modifiers-exist-p (key)
+             (and
+              (or (not (key-meta key)) (modifiers-meta *modifiers*))
+              (or (not (key-alt key)) (modifiers-alt *modifiers*))
+              (or (not (key-hyper key)) (modifiers-hyper *modifiers*))
+              (or (not (key-super key)) (modifiers-super *modifiers*))))
            (grabit (w key)
              (loop for code in (multiple-value-list (xlib:keysym->keycodes *display* (key-keysym key))) do
```

Comment:
We could rewrite the loop to use a when clause which would make clear that the when applies to the whole of the loop 'body' ie.
```lisp
(loop for code in ...
      when (and code (key-modifiers-exist-p key))
        do
```